reviews/NFTTrader/Reentrancy.md

```diff
@@ -0,0 +1,71 @@
+# Reentrancy
+
+## By: Arcturus
+
+## Summary
+
+- The contracts use onlyOwner modifiers throughout to control access except for select unknown circumstances.
+
+### Recommendations
+
+-
+
+## Scope
+
+```
+├── Docs.docx
+└── test
+  └── 001_deployment_test.js
+└── contracts
+  ├── AXBridge.sol
+  ├── BatchSwap.sol
+  ├── BatchVault.sol
+  ├── CKBridge.sol
+  ├── ERC1155Five.sol
+  ├── ERC1155Four.sol
+  ├── ERC1155One.sol
+  ├── ERC1155Six.sol
+  ├── ERC1155Three.sol
+  ├── ERC1155Two.sol
+  ├── ERC20Four.sol
+  ├── ERC20One.sol
+  ├── ERC20Three.sol
+  ├── ERC20Two.sol
+  ├── ERC721Four.sol
+  ├── ERC721One.sol
+  ├── ERC721Three.sol
+  ├── ERC721Two.sol
+  ├── Pausable.sol
+  └── TradeSquads.sol
+```
+
+## File Review
+
+### `contracts/`
+
+#### `AXBridge.sol`
+
+- AXBridge is defined as Ownable with an onlyOwner modifier on its only function `bridgeSafeTransferFrom`. Although `bridgeSafeTransferFrom` makes an external function call `transferFrom`, on `bridgeSafeTransferFrom` the onlyOwner modifier appears to prevent it from being susceptible to reentrancy.
+
+#### `BatchSwap.sol`
+
+- Batchswap
+  - createSwapIntent
+    - State set after CryptoPunk interface(external function call) check on L#185 but no apparent obvious side effect possible.
+
+#### `BatchVault.sol`
+
+- sendVaultBalance ()
+  - Function can only be called by the owner (as a result of OpenZeppelin Ownable) and appears to not be susceptible to reentrancy.
+
+#### `CKBridge.sol`
+
+- `bridgeSafeTransferFrom` can be called by anyone with `CKInterface.transfer()` unclear why. Re-entrancy Guard recommended or another modifier/guard (such as onlyOwner)
+
+#### ERC1155Five.sol - ERC721Two.sol
+
+- Contracts appear to be test contracts implementing standard ERC20, ERC721, a and ERC1155 OpenZeppelin functionality. No external function calls made, awardItem on ERC721 and ERC1155 can be called by any user.
```

Comment:
i think in case of testing , the definition of the `awarditem` function without `reentrancyModifier` makes sense , but as it has  another parameter like tokenURI , which can refer to the malicious storage address , thus  might be an possible attack payload vector